pkg/utils/k8sutil.go

```diff
@@ -312,3 +314,20 @@ func GetSecretsAsLocalObjectReference(secrets ...string) []v1.LocalObjectReferen
 	}
 	return res
 }
+
+// GetClusterDomain returns Kubernetes cluster domain, default to "cluster.local"
+func getClusterDomain() string {
+	apiSvc := "kubernetes.default.svc"
+
+	clusterDomain := "cluster.local"
+
+	cname, err := net.LookupCNAME(apiSvc)
```

Comment:
An issue with this solution is that this function is called every time there is a new message. This could lead to a bottleneck. You will need to move the call to `getClusterDomain` outside. I would put it in the `init()` method of `kafka-consumer.go` and store it as a package variable (for example) so you don't need to re-request it.